sdk/ml/azure-ai-ml/azure/ai/ml/_telemetry/activity.py

```diff
@@ -275,8 +276,16 @@ def monitor_with_activity(
     def monitor(f):
         @functools.wraps(f)
         def wrapper(*args, **kwargs):
-            with log_activity(logger, activity_name or f.__name__, activity_type, custom_dimensions):
-                return f(*args, **kwargs)
+            tracer = logger.package_tracer if isinstance(logger, OpsLogger) else None
+            if tracer:
```

Comment:
Are there any scenarios where tracer is None (i.e failed to initialize) but logger is `OpsLogger`?

Reply:
Yes, if there's an exception triggered in `get_appinsights_log_handler`, it will return `(NullHandler, None)` which will get set as `package_logger` and `package_tracer` on the `OpsLogger` object. That's why I included the else clause below.